src/ci/scripts/run-build-from-ci.sh

```diff
@@ -18,6 +18,14 @@ echo "::add-matcher::src/ci/github-actions/problem_matchers.json"
 rustup self uninstall -y || true
 if [ -z "${IMAGE+x}" ]; then
     src/ci/run.sh
+elif [ "$IMAGE" == "sbf-solana-solana-v1" ]; then
+    sed -i -e 's/.. Default::default()/cpu: "v1".into(), \n \t\t.. Default::default()/g' \
+     compiler/rustc_target/src/spec/base/sbf_base.rs
+    src/ci/docker/run.sh sbf-solana-solana
+elif [ "$IMAGE" == "sbf-solana-solana-v2" ]; then
+    sed -i -e 's/.. Default::default()/cpu: "v2".into(), \n \t\t.. Default::default()/g' \
+     compiler/rustc_target/src/spec/base/sbf_base.rs
```

Comment:
Differently from cargo where we have the `CARGO_TARGET_<triple>_RUSTFLAGS` environment variable that allows us to pass flags to a specific target, rustc tests do not use cargo and rely on a Python tool to run.

The tool provides the `--rustc-args` argument to pass more arguments to the compiler when running tests, but it applies the CPU type to both x86 (the host where we are building the tests) and SBF. I have the same issue when using the `RUSTFLAGS` environment variable.

V1, V2, and V3 are valid CPU types for x86 in LLVM, so the tests brake since the host isn't compatible with those types.

The hack here is to bypass such a problem. I patch the sbf specification file with the CPU type and run the tests.